src/lib/server/db/session.ts

```diff
@@ -64,16 +69,15 @@ export async function validateSessionToken(token: string): Promise<SessionValida
 			return { session: null, user: null };
 		}
 		const { user, session } = result[0];
-		if (Date.now() >= session.expiresAt.getTime()) {
+		if (Date.now() > session.timestamp.getTime() + INACTIVITY_TIMEOUT) {
 			await db.delete(sessionTable).where(eq(sessionTable.id, session.id));
 			return { session: null, user: null };
-		}
-		if (Date.now() >= session.expiresAt.getTime() - 1000 * 60 * 60 * 24 * 15) {
-			session.expiresAt = new Date(Date.now() + 1000 * 60 * 60 * 24 * 30);
+		} else {
+			session.timestamp = new Date(Date.now());
```

Comment:
Use just Date.now()

Reply:
`Date.now()` returns just a number and in our database `timestamp` is a date, however, I agree, this line is bad so i will just use `new Date()`.